test/functional.js

```diff
@@ -103,17 +103,15 @@ describe('Functional tests using webpack', function() {
 
                 webpackAssert.assertOutputJsonFileMatches('entrypoints.json', {
                     main: {
-                        js: ['build/runtime.js', 'build/main.js'],
-                        css: []
+                        js: ['build/runtime.js', 'build/main.js']
                     },
                     font: {
-                        // no runtime for style entries
-                        js: [],
-                        css: ['build/font.css']
+                        js: 'build/runtime.js',
```

Comment:
I was not sure how much we wanted to alter the what the plugin was generating. I will make it so we filter out runtime if it is alone.

Reply:
Yeah, altering the output of the plugin is definitely a bit annoying and should be avoided as much as possible but I'm afraid that having `runtime.js` there would also cause the Webpack Encore Bundle to add it to the page multiple times.

Maybe wait for @weaverryan's to come by before making that change (same thing for the public path thing from another comment of mine).

Reply:
I think I'm ok with it. If you add a style entry (which you shouldn't anyways - it's a "hack" feature that's meant to help with legacy projects), then you just need to be smart enough not to *try* to render the `script` tags for that entry. If you do, yep, you may get an extra script tag for `runtime.js`. But no big deal :)

Reply:
@weaverryan It would also execute the content of that file multiple times, couldn't that be an issue?

Reply:
I have been thinking about this issue. I think the bundle could do something like array_merge and it would dedup the array of the same file. The reason I think it should be done in the bundle is because what if the user only requests one entry point and it was one of the ones that does not have runtime.js. They would not get all the files needed. Although in this case I see it only popped up because we pulled in css. So in that case we could then still do it here if we check that js count is 1 and is the runtime.js file and css count 1 or more then remove it. I will do which ever approach we want to take.

Reply:
Hm, I'm not sure your example would be an issue.
Removing `runtime.js` would only be done when it's the only element for the `js` key, which would also mean that it's not needed for the related entry anyway (no other JS file).

That being said, maybe you're right and it could be handled by the bundle instead...

Reply:
Ah I see your point now a little more clearly. Yeah I will filter it out later today and get this PR updated asap. So we can get this merged and running.

Reply:
Just to chime in here: the bundle already prevents the same file from being rendered on the same page load. So, if you have an `app` entry that (of course) includes `runtime.js` and you also render another entry - `checkout` - there will be no `script` tag for `runtime.js` for checkout. This is needed in general, because when files are split, they may be "required" for multiple entry points, but we don't want to render multiple, duplicate `script` tags on the same page.

So, `runtime.js` will always be included 0 or 1 times. And, most likely, for a style entry, it will already be included in the page due to some other entry. And even if it is *not* (meaning, there is literally no JS on this page). it's harmless: just contains bootstrap code.

Does that alleviate concerns @Lyrkan?

Reply:
It does :)